### PR TITLE
♻️ Migrate to non-nullable by default

### PR DESCRIPTION
**Pull request by AlexV525**
Reference: fluttercandies/flutter_candies#5

### CODE REVIEW

File: lib/src/editor/extended_image_editor.dart [hunk 1]
```diff
@@ -1,85 +1,102 @@
...
-    final double cropAspectRatio = _editorConfig?.cropAspectRatio;
+  void _initGestureConfig({bool isInitialize = false}) {
     _editorConfig = widget
-            ?.extendedImageState?.imageWidget?.initEditorConfigHandler
+            .extendedImageState.imageWidget.initEditorConfigHandler
             ?.call(widget.extendedImageState) ??
         EditorConfig();
-    if (cropAspectRatio != _editorConfig.cropAspectRatio) {
-      _editActionDetails = null;
+    if (isInitialize) {
```
Comment (zmtzawqlp) [inline lib/src/editor/extended_image_editor.dart:64]: don't change old logic

File: lib/src/editor/extended_image_editor.dart [hunk 2]
```diff
@@ -322,10 +339,6 @@ class ExtendedImageEditorState extends State<ExtendedImageEditor> {
   }
 
   void reset() {
-    setState(() {
-      _editorConfig = null;
-      _editActionDetails = null;
-      _initGestureConfig();
-    });
+    _initGestureConfig();
```
Comment (zmtzawqlp) [inline lib/src/editor/extended_image_editor.dart:342]: miss setState, _editActionDetails should be reset

**Comment by jerryrt** [comment]
Flutter 2.0 is out there, this is a block issue in projects which depends on extended_image. Could this merge be prioritized and released?

**Comment by AlexV525** [comment]
> Flutter 2.0 is out there, this is a block issue in projects which depends on extended_image. Could this merge be prioritized and released?

Thanks for keeping up, but we recommend you to delay a little bit, since it's a huge update and need more time to review and test.

**Comment by zmtzawqlp** [comment]
3.00 is ready on pub.dev.  Thank you very much for your contributions.

**Pull request closed without merging**